### PR TITLE
Collect set extension cherry-picking and changes as per druid 23

### PR DESCRIPTION
**Pull request by sachinsagare**
Fixes #XXXX.

#### Added a forbidden-apis entry ...

<hr>

##### Key changed/added classes in this PR
 * `MyFoo`
 * `OurBar`
 * `TheirBaz`

<hr>

This PR has:
- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: sql/src/main/java/org/apache/druid/sql/calcite/aggregation/Aggregation.java [hunk 1]
```diff
@@ -52,13 +52,13 @@ private Aggregation(
...
     this.aggregatorFactories = Preconditions.checkNotNull(aggregatorFactories, "aggregatorFactories");
-    this.postAggregator = postAggregator;
+    this.postAggregator = (postAggregator != null && postAggregator.getName() == null)? null :postAggregator;
 
     if (aggregatorFactories.isEmpty()) {
       Preconditions.checkArgument(postAggregator != null, "postAggregator must be present if there are no aggregators");
     }
-
-    if (postAggregator == null) {
+    //SSAGARE-Added postAggregator.getName in condition ass] we are passing postAgreegator with all variable as null
```
Comment (debasatwa29) [inline sql/src/main/java/org/apache/druid/sql/calcite/aggregation/Aggregation.java:60]: you can remove SSAGARE, as the commit message and commit ID should contains the full details.

File: processing/src/main/java/org/apache/druid/query/aggregation/AggregatorFactory.java [hunk 2]
```diff
@@ -239,11 +241,12 @@ public ColumnType getIntermediateType()
   public ColumnType getResultType()
   {
     // this default 'fill' method is incomplete and can at best return 'unknown' complex
-    final ValueType finalized = getFinalizedType();
-    if (finalized == ValueType.COMPLEX) {
+    //SSAGARE: commented code as below getFinalizedType() function call is depricated  therefore now returning UNKNOWN_COMPLEX value
```
Comment (debasatwa29) [inline processing/src/main/java/org/apache/druid/query/aggregation/AggregatorFactory.java:244]: you can remove SSAGARE, as the commit message and commit ID should contains the full details. The commit details will contain the responsible person for the commit.

File: processing/src/main/java/org/apache/druid/query/aggregation/AggregatorFactory.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
 import org.apache.druid.segment.ColumnInspector;
 import org.apache.druid.segment.ColumnSelectorFactory;
 import org.apache.druid.segment.column.ColumnType;
-import org.apache.druid.segment.column.ColumnTypeFactory;
+//import org.apache.druid.segment.column.ColumnTypeFactory;
```
Comment (debasatwa29) [inline processing/src/main/java/org/apache/druid/query/aggregation/AggregatorFactory.java:30]: Is that getting accepted in checkstyle?

Comment (sachinsagare) [reply processing/src/main/java/org/apache/druid/query/aggregation/AggregatorFactory.java:30]: i think there is no code style for processing tht why not failed

**Review by itallam — APPROVED** [review]

**Pull request merged**